power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -457,7 +437,14 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
 
         // calculate once to cache topology, ignore results, all math solvers are initialized
         try {
-            calculation_fn(*this, {}, ignore_output);
+            calculation_fn(*this,
+                           {
+                               false,
+                               1,
+                               "sym_output",
+                               *meta_data_,
+                           },
```

Comment:
We can't in the main model. Main model does not have access now to `meta_data_gen`. So it has to passthrough the `meta_data_`.

Reply:
it can still be some function

```cpp
constexpr auto empty_output_dataset(MetaData const* meta_data) {
    return DataSet{false, 1, "sym_output", meta_data};
}
```

in addition, it doesn't always need to be `"sym_output"`, so maybe it could even be useful to declare a separate `"empty"` dataset

Reply:
I do not see the benefit of that if you cannot embed `meta_data`. Calling that function is not shorter.